src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs

```diff
@@ -52,8 +52,8 @@ public unsafe struct NetworkInterfaceInfo
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_EnumerateGatewayAddressesForInterface")]
         public static extern unsafe int EnumerateGatewayAddressesForInterface(void* context, uint interfaceIndex, delegate* unmanaged<void*, IpAddressInfo*, void> onGatewayFound);
 
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetNetworkInterfaces")]
-        public static unsafe extern int GetNetworkInterfaces(ref int count, ref NetworkInterfaceInfo* addrs, ref int addressCount, ref IpAddressInfo *aa);
+        [GeneratedDllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetNetworkInterfaces")]
+        public static unsafe partial int GetNetworkInterfaces(ref int count, ref NetworkInterfaceInfo* addrs, ref int addressCount, ref IpAddressInfo *aa);
```

Comment:
Can't we make this one blittable?

Reply:
Sadly, no. The `NetworkInterfaceInfo` has an embedded fixed array.

Reply:
Embedded `fixed` arrays of `byte`s shouldn't stop a type from being blittable.

Reply:
Managed array fields with `[MarshalAs(UnmanagedType.ByValArray)]` aren't blittable, but `fixed byte fieldName[n];` fields are.

Reply:
Sorry. In this case the `[MarshalAs(UnmanagedType.ByValArray)]` is what is being used.

Reply:
Where in the code is it using that? I just checked and it was using `fixed` for the three arrays.

Also, the runtime wouldn't let you have a pointer type to a struct that has array fields.

Can you point me to the code that uses the ByValArray annotation?

Reply:
https://github.com/dotnet/runtime/blob/2e2d1fb06693b1c44e1fb0ea48ad980cc1cc4500/src/libraries/Common/src/Interop/Windows/IpHlpApi/Interop.NetworkInformation.cs#L238-L292

Reply:
Was commenting on the wrong method. Let me try this one again.

Reply:
That only prevents the `Interop.IpHlpApi.GetIfEntry2` API from being changed to a blittable API.

I don't see anything that stops this API, `Interop.Sys.GetNetworkInterfaces` from being changed to a blittable signature.